### DISCUSSION TITLE
Port Catppuccin icons to Nerd Fonts (for Neovim, etc)

### DISCUSSION THREAD

**Discussion by noahbald** [catppuccin/catppuccin#2238] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the suggested port be themed?

There's probably a few ways this could be done
- Create a standalone font with just the nerd glyphs which can be used as a fallback font, like may be done with font_with_fallback in Wezterm
- Create a fork of the [nerd fonts repo](https://github.com/ryanoasis/nerd-fonts) and update all the fonts to use the Catppuccin icons where available
- Create a fork of the [nerd fonts patching script](https://github.com/ryanoasis/nerd-fonts?tab=readme-ov-file#font-patcher) to use Catppuccin icons where available

### Any additional comments?

Curious for people's thoughts on this and whether it's worth working on.
I'm doubtful that a fork of icons would ever get approved for addition to the fontbase, my thought of attack here is to create a script to overwrite a user's font file.
For icons used in https://github.com/nvim-tree/nvim-web-devicons/blob/master/lua/nvim-web-devicons/icons-default.lua that overlap with Catppuccin's icon pack, overwrite the given glyphs

### Submission Guidelines

- [X] I have read and followed Catppuccin's submission guidelines.

**Comment by noahbald** [comment]
Best option is to probably to create a font with necessary glyphs that can be passed as the `--custom` option to the nerds-font script.
https://github.com/ryanoasis/nerd-fonts/wiki/ScriptOptions#expert-options

I can compile a list of which Unicode characters we'd want to replace with Catppuccin's list of svg icons, rename those icons as `<unicode>-<name>.svg`, and pass it to `npx svgicons2svgfont --fontname=catppuccin-icons -o catppuccin-icons.svg icons/*.svg` which can be used by something like svg2ttf

**Comment by noahbald** [comment]
I'll be sticking to the icons in v0.33.0 as to stay stable and to continue use of icons removed in v1.0

**Comment by noahbald** [comment]
Created WIP repo https://github.com/noahbald/catppuccin-font-icons, scripts run but font's don't seem to work properly yet

**Reply by 0xJWLabs** [reply]
Hey, have you fix it yet?

**Comment by MuntasirSZN** [comment]
@noahbald Can you check my port review pls? We can then patch it.

**Reply by noahbald** [reply]
https://github.com/MuntasirSZN/code-icons/blob/main/package.json
I’m guessing this one? I can take a look and see if it works, but just by glance you might want to use a git sub-modules like I did for the SVG icons, so they don’t need to be maintained in two repos.

**Reply by MuntasirSZN** [reply]
Actually, this repo will be the central and vscode icons and other icon based repos will be merged with vscode and others. When they get merged, they will use my repo.

**Comment by Redhawk18** [comment]
I'd love to use these, good luck.
